tests/tests_integration/test_api/test_simulators/test_models.py

```diff
@@ -0,0 +1,130 @@
+)
+class TestSimulatorModels:
+    def test_list_models(self, cognite_client: CogniteClient, seed_resource_names) -> None:
+        models = cognite_client.simulators.models.list(
+            limit=5, filter=SimulatorModelsFilter(simulator_external_ids=[seed_resource_names["simulator_external_id"]])
+        )
+
+        model_ids = []
+        # quick test of the iterator
+        for model in cognite_client.simulators.models(limit=2):
+            assert model.created_time is not None
+            model_ids.append(model.id)
+
+        found_models = cognite_client.simulators.models.retrieve_multiple(ids=model_ids)
+
+        assert len(found_models) == len(model_ids)
+
+        assert len(models) > 0
+
+    def test_retrieve_model(self, cognite_client: CogniteClient, seed_resource_names) -> None:
+        model_external_id = seed_resource_names["simulator_model_external_id"]
+        model = cognite_client.simulators.models.retrieve(external_id=model_external_id)
+        assert model is not None
+        assert model.external_id == model_external_id
+        assert model.created_time is not None
+        assert model.last_updated_time is not None
+        assert model.type is not None
+        assert model.data_set_id is not None
+        assert model.name is not None
+
+    def test_list_model_revisions(self, cognite_client: CogniteClient, seed_resource_names) -> None:
+        model_external_id = seed_resource_names["simulator_model_external_id"]
+
+        revisions = cognite_client.simulators.models.revisions.list(
+            limit=5,
+            filter=SimulatorModelRevisionsFilter(model_external_ids=[model_external_id]),
+        )
+
+        model_revision_ids = []
+        # quick test of the iterator
+        for revision in cognite_client.simulators.models.revisions(limit=2):
+            assert revision.created_time is not None
+            model_revision_ids.append(revision.id)
+
+        found_revisions = cognite_client.simulators.models.revisions.retrieve_multiple(ids=model_revision_ids)
+        assert len(found_revisions) == len(model_revision_ids)
+
+        assert len(revisions) > 0
+
+    def test_retrieve_model_revision(self, cognite_client: CogniteClient, seed_resource_names) -> None:
+        model_revision_external_id = seed_resource_names["simulator_model_revision_external_id"]
+        model_revision = cognite_client.simulators.models.revisions.retrieve(external_id=model_revision_external_id)
+        assert model_revision is not None
+        assert model_revision.model_external_id == seed_resource_names["simulator_model_external_id"]
+
+    @pytest.mark.usefixtures("seed_file", "seed_resource_names")
+    def test_create_model_and_revisions(
+        self, cognite_client: CogniteClient, seed_file: FileMetadata, seed_resource_names
+    ) -> None:
+        model_external_id_1 = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+        model_external_id_2 = datetime.datetime.now().strftime("%Y%m%d%H%M%S2")
```

Comment:
Better to use a real random identifier (we have a lot of parallel test runners, and a collision on a specific second is actually likely)
`from cognite.client.utils._text import random_string`
```suggestion
        model_external_id_1 = random_string(10)
        model_external_id_2 = random_string(10)
```